pkg/utils/boards/boards.go

```diff
@@ -0,0 +1,113 @@
+package boards
+
+import (
+	"fmt"
+	"github.com/joho/godotenv"
+	"github.com/kairos-io/kairos-agent/v2/pkg/partitioner"
+	v1 "github.com/kairos-io/kairos-agent/v2/pkg/types/v1"
+	"github.com/kairos-io/kairos-agent/v2/pkg/utils/partitions"
+	"github.com/sanity-io/litter"
+	"os"
+
+	cnst "github.com/kairos-io/kairos-agent/v2/pkg/constants"
+)
+
+// IsAnAndroidBoard returns true if the system is an Android board
+// based on checking if there is a build.prop file
+func IsAnAndroidBoard() bool {
+	// Check if we are running on an Android board
+	_, err := os.Stat("/build.prop")
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+// GetAndroidBoardModel returns the board model if the system is an Android board
+func GetAndroidBoardModel() string {
+	// Check if we are running on an Android board
+	if IsAnAndroidBoard() {
+		buildProp, err := godotenv.Read("/build.prop")
+		if err != nil {
+			return ""
+		}
+		switch buildProp["ro.product.board"] {
+		case cnst.QCS6490:
+			return cnst.QCS6490
+		default:
+			return ""
+		}
+
+	}
+
+	return ""
+}
+
+// CheckDeps checks if the dependencies are installed on the system
+func CheckDeps(runner v1.Runner) error {
```

Comment:
umm this could be dropped I think, I added it when I was unable to inject the deps in the base image directly but now I dont think it makes much sense as we only support upgrading from our "golden image" which has those deps in it